README.md

```diff
@@ -396,8 +398,25 @@ SolidQueue.on_scheduler_stop
 
 For example:
 ```ruby
-SolidQueue.on_start { start_metrics_server }
-SolidQueue.on_stop { stop_metrics_server }
+SolidQueue.on_start do |supervisor|
+  Rails.logger.info "Supervisor #{supervisor.name} started"
+
+  start_metrics_server
+end
+
+SolidQueue.on_stop do |supervisor|
+  Rails.logger.info "Supervisor #{supervisor.name} stopped"
+
+  stop_metrics_server
+end
+
+SolidQueue.on_worker_start do |worker|
+  Rails.logger.info "Worker #{worker.name} started with queues: #{worker.queues.join(',')}"
+end
+
+SolidQueue.on_worker_stop do |worker|
+  Rails.logger.info "Worker #{worker.name} stopped with queues: #{worker.queues.join(',')}"
+end
```

Comment:
I know this is just because these are examples, but Solid Queue's log subscriber already logs all this when processes start and stop 🤔  I'm just not sure if it's ok to add examples that would be discouraged because you'd end up with duplicated/redundant logs.

Reply:
I changed it to instead set a couple values in a somewhat contrived `MyMetricsReporter` class, which hopefully does a better job of showing what someone might want to use it for.